### ISSUE TITLE
Add support to register ClassDefinition using hazelcast-config.xml

### ISSUE THREAD

**Issue by hwasin** [hazelcast/hazelcast#14459]
**Version**: 3.9.4, 3.11.1

Hi. Right now its possible to register custom ClassDefinition programmatically like this:
```
hazelcastInstance.getConfig().getSerializationConfig().addClassDefinition(classDefinition);
```

but [its not possible](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/resources/hazelcast-full-example.xml) to register it using hazelcast-config.xml. 

**Motivation**: I've caught following exception: 
```
Caused by: com.hazelcast.nio.serialization.HazelcastSerializationException: Cannot write null portable without explicitly registering class definition!
	at com.hazelcast.internal.serialization.impl.ClassDefinitionWriter.writePortable(ClassDefinitionWriter.java:139)[274:com.hazelcast:3.9.4]
```

in order to avoid it, ClassDefinition for the object with null fields should be registered manually before Hazelcast instance is created and this registration could be done only in programmatically today. This environment programmatic way implementation may be sophisticated in application with OSGi environment 

Thank you.

**Comment by the-thing** [comment]
Is that something I could pick up?

**Comment by mmedenjak** [comment]
Yes, it should be simple, other than adding configuration being dull and repetitive. Maybe @sancar has other observations. Still, I don't think it's a right time to start working on this as we're about to finish a release and we're not accepting any new enhancements, only bug fixes.

**Comment by the-thing** [comment]
Ok. I will hold off for the time being. These tasks as dull as it might seem are a good way to get familiar with the codebase.
